src/topp/SageAdapter.cpp

```diff
@@ -338,20 +842,51 @@ class TOPPSageAdapter :
 
     config_file.substitute("##enzyme_details##", enzyme_details);
 
+    
     auto fixed_mods = getStringList_("fixed_modifications");
     set<String> fixed_unique(fixed_mods.begin(), fixed_mods.end());
     fixed_mods.assign(fixed_unique.begin(), fixed_unique.end());   
     ModifiedPeptideGenerator::MapToResidueType fixed_mod_map = ModifiedPeptideGenerator::getModifications(fixed_mods); // std::unordered_map<const ResidueModification*, const Residue*> val;
     String static_mods_details = getModDetailsString(fixed_mod_map);
-    config_file.substitute("##static_mods##", static_mods_details);
 
     auto variable_mods = getStringList_("variable_modifications");
     set<String> variable_unique(variable_mods.begin(), variable_mods.end());
     variable_mods.assign(variable_unique.begin(), variable_unique.end());
     ModifiedPeptideGenerator::MapToResidueType variable_mod_map = ModifiedPeptideGenerator::getModifications(variable_mods);
     String variable_mods_details = getModDetailsString(variable_mod_map);
 
-    config_file.substitute("##variable_mods##", variable_mods_details);
+    //Treat variables as list for sage v0.15 and beyond 
+
+    StringList static_mods_details_list; 
+    StringList variable_mods_details_list; 
+
+
```

Comment:
```suggestion

```

Reply:
Removed unnecessary empty lines, hopefully caught all of them. Running one last test after reformatting then pushing again.